modules/cluster/cluster.tf

```diff
@@ -16,7 +16,7 @@ resource "oci_containerengine_cluster" "k8s_cluster" {
   }
 
   endpoint_config {
-    is_public_ip_enabled = var.control_plane_is_public
+    is_public_ip_enabled = var.control_plane_is_public ? var.assign_public_ip_to_control_plane : var.control_plane_is_public
```

Comment:
you need to review the logic on this. it should be if both control_plane and assign_public_ip are true, then true, else false.

Reply:
@hyder -Fixed